### ISSUE TITLE
Timeout during bulk index restore

### ISSUE THREAD

**Issue by dwhenry** [patientslikeme/es_dump_restore#20]
We currently get the occasional timeout errors from the `_bulk` command when we are restoring the dataset. 

These are frustrating as they require the entire index to be processed when ideally we would:

* retry the bulk request at least once
* increase the timeout to avoid this issue when the index is under load

I am hoping to get some guidance before starting a PR around your preferred solution.

For the retry option, I would pass in the number of times the request can be retried for a specific error class with a default of zero so that it does change existing functionality:

```
option = { 'error_retry' => { 'TimoutCErrorlass' => 2, .... } }
```

**Comment by nbudin** [comment]
I think your proposed solution makes sense.  We might even want to ship with some reasonable defaults - retrying timeouts seems like a good one.  Just to clarify: the limit would apply to a specific request, right?  (As opposed to globally, i.e. "once there have been X timeouts in this batch operation, we'll error out"...?)

**Comment by dwhenry** [comment]
@nbudin that is correct the limit would be on a specific request. I will try and get a PR ready in the next couple of days then. :)

**Comment by nbudin** [comment]
Fixed in 2.2.0.
